pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -558,7 +573,7 @@ func (s *Controller) WorkloadInstanceHandler(wi *model.WorkloadInstance, event m
 	}
 	s.mutex.Unlock()
 
-	s.edsUpdate(instances)
+	s.edsUpdate(append(instances, instancesDeleted...))
```

Comment:
why doyou add deleted instances?

This function edsUpdate will do cache refresh

Reply:
understand now